src/provisioning/api.md

```diff
@@ -0,0 +1,290 @@
+{}
+```
+
+Any other response should be considered a failed request (e.g. 404, 502 etc).
+
+## GET /v1/connectiontypes
+
+Request the connection types enabled for this bridge.
+
+### Response
+
+```json5
+{
+    "JiraProject": {
+        "type": "JiraProject", // The name of the connection
+        "eventType": "uk.half-shot.matrix-hookshot.jira.project", // Corresponds to the state type for the connection
+        "service": "jira", // or github, webhook. A human-readable service name to make things look pretty
+        "botUserId": "@hookshot:yourdomain.com", // The bot mxid for the service. Currently this is the sender_localpart, but may change in the future.
+    }
+}
+```
+
+## GET /v1/{roomId}/connections
+
+Request the connections for a given room. The `{roomId}` parameter is the target Matrix room.
+
+### Response
+
+```json5
+[{
+    "type": "JiraProject", // The name of the connection
+    "eventType": "uk.half-shot.matrix-hookshot.jira.project", // Corresponds to the state type in the connection
+    "id": "opaque-unique-id", // An opaque ID used to refer to this connection. Should **NOT** be assumed to be stable.
+    "service": "jira", // or github, webhook. A human-readable service name to make things look pretty
+    "botUserId": "@hookshot:yourdomain.com", // The bot mxid for the service. Currently this is the sender_localpart, but may change in the future.
+    "config": {
+        // ... connection specific details, can be configured.
+    }
+}]
+```
+
+
+## GET /v1/{roomId}/connections/{id}
+
+Request details of a single connection. The `{roomId}` parameter is the target Matrix room.
+
+### Response
+
+```json5
+{
+
+    "type": "JiraProject", // The name of the connection
+    "eventType": "uk.half-shot.matrix-hookshot.jira.project", // Corresponds to the state type in the connection
+    "id": "opaque-unique-id", // An opaque ID used to refer to this connection. Should **NOT** be assumed to be stable.
+    "service": "jira", // or github, webhook. A human-readable service name to make things look pretty
+    "botUserId": "@hookshot:yourdomain.com", // The bot mxid for the service. Currently this is the sender_localpart, but may change in the future.
+    "config": {
+        // ... connection specific details, can be configured.
+    }
+}
+```
+
+## PUT /v1/{roomId}/connections/{type}
+
+Create a new connection of a given type. The type refers to the `IConnection.CanonicalEventType`. The `{roomId}` parameter is the target Matrix room.
+
+The body of the request is the configuration for the connection, which will be the "ConnectionState" interface for each connection.
+
+The request will respond with a `202` on success, as the connection creation process is asyncronous (being driven by Matrix state).
+
+### Request body
+```json5
+{
+    // ... connection specific details, can be configured.
+}
+```
+### Response
+
+```json5
+{
+    // The eventId of the state event that describes the connection.
+    "eventId": "!abc:def"
```

Comment:
echoing the connection back to the caller is a bit more useful than the event ID. With an event ID, the caller needs to have context of the room which might not be possible (particularly in the case of integration managers) whereas the connection information means that the caller can simply append the response to a local array instead of having to re-fetch all connections again.

Reply:
This was a bit difficult due to our asynchronous processing of new connections, but I have hopefully solved this now.